netmaster/k8snetwork/networkpolicy.go

```diff
@@ -1,36 +1,114 @@
 package networkpolicy
 
 import (
-	"encoding/json"
 	"fmt"
-	"reflect"
-	"strings"
-	"time"
-
 	log "github.com/Sirupsen/logrus"
 	"github.com/contiv/netplugin/contivmodel/client"
 	"github.com/contiv/netplugin/utils/k8sutils"
-	"k8s.io/api/networking/v1"
+	v1 "k8s.io/api/core/v1"
+	network_v1 "k8s.io/api/networking/v1"
 	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/kubernetes"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
 )
 
 const defaultTenantName = "default"
-const defaultNetworkName = "net"
+const defaultNetworkName = "default-net"
 const defaultSubnet = "10.1.2.0/24"
-const defaultEpgName = "ingress-group"
+
+//const defaultEpgName = "ingress-group"
+//const defaultEpgName = "default-epg"
+const defaultEpgName = "default-group"
 const defaultPolicyName = "ingress-policy"
 const defaultRuleID = "1"
-
+const defaultPolicyPriority = 2
+
+type k8sPodSelector struct {
+	TenantName  string //Attach Tenant
+	NetworkName string //Attach network
+	GroupName   string //Attach EPG
+	PolicyName  string //Attach to policy
+	labelPodMap map[string]map[string]bool
+	podIps      map[string]string
+}
+type k8sPolicyPorts struct {
+	Port     int
+	Protocol string
+}
+type k8sNameSelector struct {
+	nameSpaceSel string
+}
+type podCache struct {
+	labeSelector string
+	podNetwork   string
+	podGroup     string
+	podEpg       string
+}
+type k8sIPBlockSelector struct {
+	subNetIps []string
+}
+type k8sIngress struct {
+	IngressRules           []k8sPolicyPorts
+	IngressPodSelector     []*k8sPodSelector
+	IngressNameSelector    *k8sNameSelector
+	IngressIpBlockSelector *k8sIPBlockSelector
+}
+type npPodInfo struct {
+	nameSpace      string
+	labelSelectors []string
+	IP             string //??? should care for ipv6 address
+}
+type k8sNetworkPolicy struct {
+	PodSelector *k8sPodSelector
+	Ingress     []k8sIngress
+}
+type labelPolicySpec struct {
+	policy []*k8sNetworkPolicy
+}
 type k8sContext struct {
 	k8sClientSet *kubernetes.Clientset
 	contivClient *client.ContivClient
 	isLeader     func() bool
+	//Policy Obj per  Policy Name
+	networkPolicy map[string]*k8sNetworkPolicy
+	//List of Rules Per Policy
+	policyRules map[string][]string
+	//List of Network configured
+	network map[string]string
+	//List of EPG configured
+	epgName map[string]string
+	//Default  policy Per EPG
+	defaultPolicyPerEpg map[string]string
+	//List of Policy Per EPG
+	policyPerEpg map[string][]string
+	//Cache table for given Pods
+	//Policy Obj per  Policy Name
+	nwPolicyPerNameSpace map[string][]*k8sNetworkPolicy
 }
 
 var npLog *log.Entry
 
+func getNetworkInfo() string {
+	//XXX: Need expend this in version 2
+	return defaultNetworkName
+}
+func getLabelDBkey(label, nameSpace string) string {
+	return label + nameSpace
+}
+func getEpgInfo() string {
+	//XXX: Need to expend in version 2
+	return defaultEpgName
+}
+func getTenantInfo() string {
+	return defaultTenantName
```

Comment:
why these 4 functions are required ?

Reply:
Initially thinking to use wrapper function to access default tenant and EPG name . look like currently not using it. will remove it